test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -3809,7 +3811,7 @@ public void PackCommand_PackWithSourceControlInformation_PublishedUrl_VerifyNusp
                     var repositoryMetadata = nuspecReader.GetRepositoryMetadata();
                     repositoryMetadata.Type.Should().Be("git");
                     repositoryMetadata.Url.Should().Be("https://github.com/NuGet/NuGet.Client.git");
-                    repositoryMetadata.Branch.Should().Be("");
+                    repositoryMetadata.Branch.Should().Be("refs/heads/main");
```

Comment:
Do we have any idea why branch wasn't set before?
I'm asking cause there's an explicit assertion in this test.

Reply:
I think it was to codify the previous SourceLink behavior. L3639 is an example: if the input property is set it shows up in the .nuspec (see https://github.com/NuGet/NuGet.Client/blob/60974f11275eabfdfb364a76130aded590cb144d/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs#L3639). These tests we're asserting that if it wasn't explicitly set, it was empty.

Now these tests assert that if it isn't explicitly set, the value from SourceLink is used.

/cc @tmat to correct me if I'm wrong.

Reply:
Yup, just validation of an existing behavior.